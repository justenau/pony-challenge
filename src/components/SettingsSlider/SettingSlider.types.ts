export interface SettingsSliderProps {
    setting: string;
    min: number;
    max: number;
    value: number;
    handleChange: any;
}
