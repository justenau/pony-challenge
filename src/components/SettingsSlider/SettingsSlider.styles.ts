import { makeStyles } from '@material-ui/styles';

export default makeStyles(() => ({
    container: {
        width: '70%',
    },

    setting: {
        display: 'flex',
        flexDirection: 'row',
        padding: '0 1em',
        alignItems: 'center',
    },

    label: {
        width: '30%',
        textAlign: 'left',
        paddingRight: '10px',
    },
}));
