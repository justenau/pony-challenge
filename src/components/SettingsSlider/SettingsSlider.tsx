import React from 'react';
import { Slider } from '@material-ui/core';
import Typography from '@material-ui/core/Typography';
import useStyles from './SettingsSlider.styles';
import { SettingsSliderProps } from './SettingSlider.types';

export const SettingsSlider: React.FC<SettingsSliderProps> = ({ setting, min, max, value, handleChange }) => {
    const classes = useStyles();

    return (
        <div className={classes.setting}>
            <Typography className={classes.label} variant={'body2'}>
                {setting}
            </Typography>
            <Slider
                value={value}
                onChange={handleChange}
                marks={[{ value: min }, { value: max }]}
                step={1}
                valueLabelDisplay='auto'
                min={min}
                max={max}
            />
        </div>
    );
};
