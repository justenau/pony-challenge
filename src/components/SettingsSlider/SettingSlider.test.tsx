import { render } from '@testing-library/react';
import React from 'react';
import { SettingsSlider } from './SettingsSlider';

describe('Test SettingsSlider component', () => {
    it('should match snapshot', () => {
        const tree = render(<SettingsSlider setting={'width'} min={15} max={25} value={20} handleChange={jest.fn()} />);
        expect(tree).toMatchSnapshot();
    });
});
