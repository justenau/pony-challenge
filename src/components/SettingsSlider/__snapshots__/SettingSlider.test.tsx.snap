// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SettingsSlider component should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-setting-2"
      >
        <p
          class="MuiTypography-root makeStyles-label-3 MuiTypography-body2"
        >
          width
        </p>
        <span
          class="MuiSlider-root MuiSlider-colorPrimary"
        >
          <span
            class="MuiSlider-rail"
          />
          <span
            class="MuiSlider-track"
            style="left: 0%; width: 50%;"
          />
          <input
            type="hidden"
            value="20"
          />
          <span
            class="MuiSlider-mark MuiSlider-markActive"
            style="left: 0%;"
          />
          <span
            aria-hidden="true"
            class="MuiSlider-markLabel MuiSlider-markLabelActive"
            style="left: 0%;"
          />
          <span
            class="MuiSlider-mark"
            style="left: 100%;"
          />
          <span
            aria-hidden="true"
            class="MuiSlider-markLabel"
            style="left: 100%;"
          />
          <span
            aria-orientation="horizontal"
            aria-valuemax="25"
            aria-valuemin="15"
            aria-valuenow="20"
            class="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-54"
            data-index="0"
            role="slider"
            style="left: 50%;"
            tabindex="0"
          >
            <span
              class="PrivateValueLabel-offset-56 MuiSlider-valueLabel"
            >
              <span
                class="PrivateValueLabel-circle-57"
              >
                <span
                  class="PrivateValueLabel-label-58"
                >
                  20
                </span>
              </span>
            </span>
          </span>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-setting-2"
    >
      <p
        class="MuiTypography-root makeStyles-label-3 MuiTypography-body2"
      >
        width
      </p>
      <span
        class="MuiSlider-root MuiSlider-colorPrimary"
      >
        <span
          class="MuiSlider-rail"
        />
        <span
          class="MuiSlider-track"
          style="left: 0%; width: 50%;"
        />
        <input
          type="hidden"
          value="20"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive"
          style="left: 0%;"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-markLabel MuiSlider-markLabelActive"
          style="left: 0%;"
        />
        <span
          class="MuiSlider-mark"
          style="left: 100%;"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-markLabel"
          style="left: 100%;"
        />
        <span
          aria-orientation="horizontal"
          aria-valuemax="25"
          aria-valuemin="15"
          aria-valuenow="20"
          class="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-54"
          data-index="0"
          role="slider"
          style="left: 50%;"
          tabindex="0"
        >
          <span
            class="PrivateValueLabel-offset-56 MuiSlider-valueLabel"
          >
            <span
              class="PrivateValueLabel-circle-57"
            >
              <span
                class="PrivateValueLabel-label-58"
              >
                20
              </span>
            </span>
          </span>
        </span>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
